core/src/main/java/org/elasticsearch/tribe/TribeService.java

```diff
@@ -352,8 +375,7 @@ public String describeTasks(List<ClusterChangedEvent> tasks) {
             BatchResult.Builder<ClusterChangedEvent> builder = BatchResult.builder();
 
             try {
-                // we only need to apply the latest cluster state update
-                accumulator = applyUpdate(accumulator, tasks.get(tasks.size() - 1));
+                accumulator = applyUpdate(accumulator, tasks);
```

Comment:
this method became useless I think? the whole try catch can be removed as it's handled by the caller in the `ClusterService`. I think we should break applyUpdate into several methods each updating a different parts. i.e., updateNodes, updateIndices and now the now updateCustoms